### ISSUE TITLE
Public Beta

### ISSUE THREAD

**Issue by tobiasKaminsky** [owncloud/android#1213]
In order to pre-test the PRs by the community I suggest to create a public beta in a new branch.
There will be an apk, that experienced user can download, install and test.
This branch will contains all PRs that are "3 - to review" or above.
Issues regarding the beta will get the label "Beta".
Found bugs or improvements are implemented in the corresponding original PRs and merged back to beta.
The main advantage of this is, that the PRs are tested by far more user than "only" the QA. 
Furthermore if the PRs have fewer issues, the workload of the QA should be lower and thus can result in a faster merge process.

I am willing to maintain this branch and do a merge if the belonging PRs are changing.
What do you think? @davivel @rperezb @jancborchardt @AndyScherzinger @LukeOwncloud

**Comment by jancborchardt** [comment]
I’d say this would be very good to ensure a good quality level and maybe even get some people involved in development.

**Comment by LukeOwncloud** [comment]
Great idea! And thank you for your offer to act as maintainer! I'd love to test that beta.

About creating the APK: It should always the signed with the same key to allow updating without prior deinstalling. Will you, @tobiasKaminsky, always create it or should we create a "shared beta key"?

**Comment by tobiasKaminsky** [comment]
Ah. Good point about the signing of the app.
I guess it is not a good idea to expose the "beta key" for security reasons, or?
So, I would do this. 
In fact it is not that often, I think: Once a new PR is ready for testing and once a bug/enhancement was committed. So I guess roughly max 2-3 per week...
This is something I can do :)

**Comment by AndyScherzinger** [comment]
I think this is a good idea! I also always work run my latest PR on my phone to see if everything works fine over time. Having other people and/or devs running the version would provide valuable feedback :) :+1:

**Comment by xylo** [comment]
Great idea. I would also like to test the new features.
If there is interest I would offer to set up an f-droid repository for the betas (in case you didn't have the same plan yet).

**Comment by tobiasKaminsky** [comment]
Please see https://github.com/owncloud/android/tree/beta
I have updated the Readme and Changelog.
The next days I will add other PRs, sorted from old to new, to get the old ones tested (and merged) sooner.
Currently there is a compile error by travis, which I will fix tomorrow.

Happy testing and please use the label "Beta" if finding bugs, or comment in the corresponding PR.

**Comment by rperezb** [comment]
Thanks a lot @tobiasKaminsky for your willingness!

**Comment by jancborchardt** [comment]
Very important here is that the beta uses a different ID than the normal app, so that they can be installed side by side. Is that possible @tobiasKaminsky @rperezb?

**Comment by tobiasKaminsky** [comment]
@jancborchardt I have included this. We can now install the app side by side :)

**Comment by tobiasKaminsky** [comment]
Ladies and Gentleman: We now have a nearly up-to-date beta version with every PR labeled "3 - ready to review".
If you install it, please leave a short note here. In that way we can see how many beta users we have.

**Comment by tobiasKaminsky** [comment]
https://github.com/owncloud/android/blob/beta/apks/owncloud-beta-2015-10-29.apk
Here is a direct link to it :)

**Comment by tobiasKaminsky** [comment]
@all Please feel free to ping me, if you see that a PR is missing or a PR has been updated and should be included in the beta.

**Comment by AndyScherzinger** [comment]
I got one (which you probably don't want to add :grin:) --> #1220 (23 as compile target
and one which might end up in the master branch soon anyways: #1192

**Comment by LukeOwncloud** [comment]
I like @xylo's offer. Thank you! This might bring us some new test users and helps test user's to stay up-to-date. 

@xylo for F-Droid, it should be clear that it is a beta version which can be installed alongside the original ownCloud app and should only be installed if the user is willing to file bug reports -> https://github.com/owncloud/android/issues
But before uploading please wait until issue https://github.com/owncloud/android/issues/1225 is fixed.

Side note: The original ownCloud app is available at F-Droid, too.

**Comment by tobiasKaminsky** [comment]
@AndyScherzinger Can you maybe provide an icon with "beta" in the lower right side?
(I am not that good with graphics...)

**Comment by supportreq** [comment]
@tobiasKaminsky @AndyScherzinger I m not sure if this is an issue with the Beta app, but if I touch any file it doesn't open the file details or start downloading the file automatically.. it is just stuck.. only images open up..

**Comment by tobiasKaminsky** [comment]
@supportreq You can verify this by using the "normal" app and if the problem only occurs in beta version, please open a new issue with the label "beta".

**Comment by LukeOwncloud** [comment]
@tobiasKaminsky thanks you a lot for this initiative! Works great. Just one more request: When providings links to the apk, please point them to the raw version which can be downloaded directly. (Downloading the other links on Android is troublesome). We can close this issue now, can't we?

**Comment by tobiasKaminsky** [comment]
I thought I have, at least "raw=true" is added to the url in https://github.com/tobiasKaminsky

**Comment by tobiasKaminsky** [comment]
You are right. Regarding the owncloud logo I will open up a new issue.

**Comment by xylo** [comment]
@LukeOwncloud Regarding the f-droid repository i opened a new issue (#1229).  I think I can work on it this weekend.
